### PR TITLE
minor bug fixes

### PR DESCRIPTION
**Pull request by jemunro**

### CODE REVIEW

File: R/processReads.R [hunk 1]
```diff
@@ -207,12 +207,12 @@ mergeAmpliconReads <- function(fastqFileR1, fastqFileR2, outputDir, mergePrefix=
     msg <- paste("Processing file", basename(fastqFileR1[i]), "and", basename(fastqFileR2[i]), "...")
     progressReport(detail=msg, value=i)
     
-    outputFile <- file.path(outputDir, sub("_F\\.fastq.gz", "", basename(fastqFileR1[i])))
+    outputFile <- file.path(outputDir, sub("_R1.fastq.gz$", "", basename(fastqFileR1[i])))
```
Comment (lerch-a) [inline R/processReads.R]: Hm, the output file should not have _R1 nor _F postfix.

File: R/callHaplotype.R [hunk 2]
```diff
@@ -284,7 +283,7 @@ createFinalHaplotypTable <- function(outputDir, sampleTable, markerTable, refere
       do.call(rbind, lapply(split(tab, tab$SampleID), function(tt){
         chim <- NULL
         hIdx <- grep(marker, tt$Haplotype)
-        if(length(hIdx)>2){
+        if(length(hIdx)>2) {
```
Comment (lerch-a) [inline R/callHaplotype.R]: I don't like the space here. Can you make sure not to introduce such whitespace character.

File: R/processReads.R (hunk 1 above)
Comment (lerch-a) [reply R/processReads.R]: I checked that and the old version with 'sub("_F\\.fastaq.gz" ...' is correct. Why do you think this is a bug?

File: R/callHaplotype.R [hunk 3]
```diff
@@ -272,9 +271,9 @@ createFinalHaplotypTable <- function(outputDir, sampleTable, markerTable, refere
     # check replicates
     idx <- split(1:dim(haplotypesSample)[2], samTab$SampleID)
     markerRes <- lapply(idx, function(i){
-      reads <- callHaplotype(haplotypesSample[,i, drop=F], minHaplotypCoverage=minHaplotypCoverage, 
-                           minReplicate=minReplicate, detectability=detectability, minSampleCoverage=minSampleCoverage, 
-                           reportBackground=T)
+      tab <- callHaplotype(haplotypesSample[,i, drop=F], minHaplotypCoverage=minHaplotypCoverage,
```
Comment (lerch-a) [inline R/callHaplotype.R:274]: good catch! I was really tired on Sunday.

**Review by lerch-a — CHANGES_REQUESTED** [review]
In processReads.R the changes from 
outputFile <- file.path(outputDir, sub("_F\\.fastq.gz", "", basename(fastqFileR1[i])))
to 
outputFile <- file.path(outputDir, sub("_R1\\.fastq.gz$", "", basename(fastqFileR1[i])))
are wrong. The files have a "_F" postfix after removePrimer! Reverse the this change and then I accept the pull request.

File: R/processReads.R (hunk 1 above)
Comment (jemunro) [reply R/processReads.R]: My mistake - I am working on some code for running HaplotypR in parallel by splitting fastq files, and I had a different extension on some of them.
For greater flexibility and consistency with demultiplexByMarker I have added the trimFilenameExt argument with default value "_F\\\\.fastq.gz$"

**Comment by jemunro** [comment]
changes implemented.

**Comment by lerch-a** [comment]
Thanks.

**Pull request merged**